### PR TITLE
Content-Range is incorrectly specified.

### PR DESCRIPTION
**Pull request by yuki2006**
For example, a file of 22709217 bytes content is

should be `0-22709216/22709217` instead of `0-22709217/22709217`.

Chrome will not handle this correctly and videos and other files will not play.

image

RFC:
https://httpwg.org/specs/rfc7233.html#header.content-range

### CODE REVIEW

**Comment by fsouza** [comment]
Thanks for contributing. Would you be able to add a test that fails with the previous code and passes with the fix. Ideally something using the SDK rather than asserting on the value of the headers.

**Comment by yuki2006** [comment]
@fsouza thx comment.

I don't know how to use the SDK, but I wrote some unit tests.

Please feel free to make any changes.

File: fakestorage/object.go [hunk 1]
```diff
@@ -686,7 +687,7 @@ func (s *Server) handleRange(obj Object, r *http.Request) (ranged bool, start in
 				if end > int64(len(obj.Content)) {
 					end = int64(len(obj.Content))
 				}
-				return true, start, end, obj.Content[start:end]
+				return true, start, end - 1, obj.Content[start:end]
```
Comment (StevenACoffman) [inline fakestorage/object.go:690]: Shouldn't this be:
```
return true, start, end - 1, obj.Content[start:end-1]
```
?

Comment (fsouza) [reply fakestorage/object.go:690]: Nope, in the slice expression the upper bound is non-inclusive: https://go.dev/ref/spec#Slice_expressions

**Pull request merged**